### PR TITLE
Add a prompt before you submit a crash report.

### PR DESCRIPTION
**Pull request by tikurahul**
Addresses: https://github.com/tessel/t2-cli/issues/745

After this change:

```
➜  CLI git:(master) t2 crash-reporter --test
ERR! Detected CLI crash [Error: Testing the crash reporter] Error: Testing the crash reporter
    at CrashReporter.off.CrashReporter.on.CrashReporter.sanitize.CrashReporter.sanitize.redactions.CrashReporter.prompt.CrashReporter.submit.CrashReporter.prompt.then.CrashReporter.post.CrashReporter.status.CrashReporter.test (/Users/rahulrav/Workspaces/Tessel2/CLI/lib/crash-reporter.js:177:25)
    at process._tickCallback (node.js:406:9)
    at Function.Module.runMain (module.js:449:11)
    at startup (node.js:141:18)
    at node.js:933:3
INFO Detected a crash in the CLI. Submit crash (y/n) ?
   If yes(y), subsequent crashes will be submitted automatically.
n
INFO Did not submit crash report.
```

### CODE REVIEW

**Comment by rwaldron** [comment]
User input things should be uniformly implemented with `controller.menu` (which is a wrapper for `inquirer`).

- [controller.menu](https://github.com/tessel/t2-cli/blob/master/lib/controller.js#L995-L1017)
- [example T2 selection](https://github.com/tessel/t2-cli/blob/master/lib/controller.js#L219-L249)

Basically this: 

```js
controller.menu({
  prompt: {
    name: 'selected',
    type: 'confirm',
    message: CRASH_PROMPT_MESSAGE,
  }
}).then(selection => {
    console.log(selection);
});
```


But you'll probably have to put `controller.menu` in a different module now, let's call it `menu.js` and then update both `lib/controller.js` and `lib/crash-reporter.js`

**Comment by johnnyman727** [comment]
@tikurahul I think the output could have more info about what submitting a crash report actually means. Why would the user want to do it? When would they not?

**Comment by tikurahul** [comment]
Yes, I agree. I can add a blurb about how crash reports help us improve the quality of the CLI. 

Do you have any suggestions on what the blurb should state ?

**Comment by johnnyman727** [comment]
Perhaps "Submit Crash Report to help Tessel Developers improve the CLI?"

Also, it looks like we print out "Detected a Crash in the CLI" twice - we could probably strip that from the submission prompt.

**Comment by tikurahul** [comment]
Sounds good. Will clean up the logs and add the blurb.

**Comment by tikurahul** [comment]
Interestingly enough, `sinon.spy` is not able to deal with modules which are pure exported functions. 

For e.g.

```javascript
var Module = (input) => { ... };
module.exports = Module;
```
In tests, `sinon.spy(Module, function() { });` causes unresolved symbol issues.

File: test/unit/menu.js [hunk 1]
```diff
@@ -78,7 +79,7 @@ exports['controller.menu'] = {
       });
     });
 
-    controller.menu({
+    Menu.prompt({
       prompt: {
         name: 'selected',
         type: 'list',
```
Comment (rwaldron) [inline test/unit/menu.js]: Very clean.

**Pull request merged**